processors/filterprocessor/gen/generate.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"os"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/algorand/go-algorand/data/transactions"
+)
+
+// recursiveTagFields recursively gets all field names in a struct
+// Output will contain a key of the full tag along with the fully qualified struct
+func recursiveTagFields(theStruct interface{}, Output *map[string]string, tagLevel []string, fieldLevel []string) {
+	rStruct := reflect.TypeOf(theStruct)
+	numFields := rStruct.NumField()
+	for i := 0; i < numFields; i++ {
+		field := rStruct.Field(i)
+		name := field.Name
+
+		var tagValue string
+		var foundTag bool
+		// If there is a codec tag...
+		if tagValue, foundTag = field.Tag.Lookup("codec"); foundTag {
+
+			vals := strings.Split(tagValue, ",")
+			// Get the first value (the one we care about)
+			tagValue = vals[0]
+			// If it is empty ignore it
+			if tagValue == "" {
+				continue
+			}
+
+			fullTag := strings.Join(append(tagLevel, tagValue), ".")
+			(*Output)[fullTag] = strings.Join(append(fieldLevel, name), ".")
+		}
+
+		if field.Type.Kind() == reflect.Struct {
+			var passedTagLevel []string
+			if foundTag {
+				passedTagLevel = append(tagLevel, tagValue)
+			} else {
+				passedTagLevel = tagLevel
+			}
+			recursiveTagFields(reflect.New(field.Type).Elem().Interface(), Output, passedTagLevel, append(fieldLevel, name))
+		}
+	}
+}
+
+// usage:
+// go run generate.go packagename outputfile
+func main() {
+
+	var packageName string
+	var outputFilepath string
+
+	if len(os.Args) == 3 {
+		packageName = os.Args[1]
+		outputFilepath = os.Args[2]
+	}
```

Comment:
should there be and `else` case here that exits gracefully?

Reply:
The way the logic works is that we build up the "string" (of which only needs the file path).  Then at the end we decide if we print to stdout or the file itself.  The reason for that was running the generate command by itself will allow you to see its output without you having to run "go generate" over and over.